File: src/07-codegolf-easy/don't-panic/solution.py
h=input().split();e={h[3]:h[4]}
exec("f,p=input().split();e[f]=p\n"*int(h[7]))
while 1:f,p,d=input().split();t=e.get(f,p);print("WAIT"if f<"0"or(d>"N")==(int(p)<int(t))or p==t else"BLOCK")

File: src/07-codegolf-easy/don't-panic/solution.cpp
#include<cstdio>
#include<map>
using namespace std;map<int,int>e;int main(){int n,w,r,ef,ep,t,c,ne;scanf("%d%d%d%d%d%d%d%d",&n,&w,&r,&ef,&ep,&t,&c,&ne);e[ef]=ep;for(;ne--;){int f,p;scanf("%d%d",&f,&p);e[f]=p;}for(;;){int f,p;char d[9];scanf("%d%d%s",&f,&p,d);int g=e.count(f)?e[f]:p;puts(f<0||p==g||(d[0]>'N')==(p<g)?"WAIT":"BLOCK");}}